amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -1090,6 +1072,31 @@ private void resetInfoLogging() {
         }
     }
 
+    private ShardInfo constructShardInfoFromLease(final Lease lease) {
+        final boolean isMultiStreamLease = lease instanceof MultiStreamLease;
+
+        final Optional<String> streamIdentifierSerialization = isMultiStreamLease ?
+                Optional.of(((MultiStreamLease) lease).streamIdentifier()) : Optional.empty();
+        final StreamConfig streamConfig = getOrCreateStreamConfig(streamIdentifierSerialization);
+
+        final String shardId = isMultiStreamLease ? ((MultiStreamLease) lease).shardId() : lease.leaseKey();
+        return new ShardInfo(
+                shardId, lease.concurrencyToken().toString(), lease.parentShardIds(), lease.checkpoint(), streamConfig);
+    }
+
+    private StreamConfig getOrCreateStreamConfig(final Optional<String> streamIdentifierSerialization) {
+        if (!streamIdentifierSerialization.isPresent()) {
+            Validate.isTrue(!isMultiStreamMode, "Should not be in MultiStream Mode");
+            final StreamConfig streamConfig = currentStreamConfigMap.values().iterator().next();
+            Validate.notNull(streamConfig, "StreamConfig should not be null");
+            return streamConfig;
+        }
+
+        final StreamIdentifier streamIdentifier =
+                StreamIdentifier.multiStreamInstance(streamIdentifierSerialization.get());
+        return currentStreamConfigMap.getOrDefault(streamIdentifier, streamTracker.createStreamConfig(streamIdentifier));
+    }
```

Comment:
```suggestion
    private ShardInfo constructShardInfoFromLease(final Lease lease) {
        Optional<MultiStreamLease> msl = Optional.of(lease).filter(l->l instanceof MultiStreamLease).map(l->(MultiStreamLease) l);
        String shardId = msl.map(MultiStreamLease::shardId).orElse(lease.leaseKey());
        StreamConfig streamConfig = msl.map(MultiStreamLease::streamIdentifier).map(streamIdentifierSerialization -> {
            final StreamIdentifier streamIdentifier = StreamIdentifier.multiStreamInstance(streamIdentifierSerialization);
            return currentStreamConfigMap.getOrDefault(streamIdentifier, streamTracker.createStreamConfig(streamIdentifier));
        }).orElseGet(()->{
            Validate.isTrue(!isMultiStreamMode, "Should not be in MultiStream Mode");
            return Optional.ofNullable(currentStreamConfigMap.values().iterator().next()).get();
        });
        return new ShardInfo(
                shardId, lease.concurrencyToken().toString(), lease.parentShardIds(), lease.checkpoint(), streamConfig);
    }
```
This makes it a little more clear that the empty optional represents an uncastable Lease.